File: src/main/java/toast/pine/Adapters/ItemMaterialAdapter.java
package toast.pine.Adapters;

import org.bukkit.persistence.PersistentDataAdapterContext;
import org.bukkit.persistence.PersistentDataType;
import org.jetbrains.annotations.NotNull;
import toast.pine.Classes.Items.ItemMaterial;

public class ItemMaterialAdapter implements PersistentDataType<ItemMaterial, ItemMaterial> {

    @Override
    public @NotNull Class<ItemMaterial> getPrimitiveType() {
        return ItemMaterial.class;
    }

    @Override
    public @NotNull Class<ItemMaterial> getComplexType() {
        return ItemMaterial.class;
    }

    @Override
    public @NotNull ItemMaterial toPrimitive(@NotNull ItemMaterial item, @NotNull PersistentDataAdapterContext persistentDataAdapterContext) {
        return item;
    }

    @Override
    public @NotNull ItemMaterial fromPrimitive(@NotNull ItemMaterial item, @NotNull PersistentDataAdapterContext persistentDataAdapterContext) {
        return item;
    }
}

File: src/main/java/pine/toast/Adapters/ItemMaterialAdapter.kt
package toast.pine.Adapters

import org.bukkit.persistence.PersistentDataAdapterContext
import org.bukkit.persistence.PersistentDataType
import toast.pine.Classes.Items.ItemMaterial


class ItemMaterialAdapter : PersistentDataType<ItemMaterial, ItemMaterial> {
    @get:JvmName("getPrimitiveTypeItemMaterial")
    val primitiveType: Class<ItemMaterial>
        get() = ItemMaterial::class.java
    @get:JvmName("getComplexTypeItemMaterial")
    val complexType: Class<ItemMaterial>
        get() = ItemMaterial::class.java

    override fun toPrimitive(
        item: ItemMaterial,
        persistentDataAdapterContext: PersistentDataAdapterContext
    ): ItemMaterial {
        return item
    }

    override fun fromPrimitive(
        item: ItemMaterial,
        persistentDataAdapterContext: PersistentDataAdapterContext
    ): ItemMaterial {
        return item
    }

    override fun getPrimitiveType(): Class<ItemMaterial> {
        return primitiveType
    }

    override fun getComplexType(): Class<ItemMaterial> {
        return complexType
    }
}